consensus/types/src/execution_proof_subnet_id.rs

```diff
@@ -0,0 +1,139 @@
+//! Identifies each execution proof subnet by an integer identifier.
+use serde::{Deserialize, Serialize};
+use ssz::{Decode, DecodeError, Encode};
+use std::fmt::{self, Display};
+use std::ops::{Deref, DerefMut};
+
+/// Maximum number of execution proof subnets allowed by the protocol.
+///
+/// This is a hard protocol limit that defines the total number of proof subnets
+/// that can exist in the network. Individual nodes may choose to participate in
+/// fewer subnets (configured via max_execution_proof_subnets in ChainConfig),
+/// but no node can exceed this protocol maximum.
+///
+/// The value of 8 subnets provides a good balance between:
+/// - Proof diversity (multiple independent proofs per block)
+/// - Network overhead (not too many gossip topics)
+/// - Resource requirements (reasonable for most nodes(?))
+///
+/// In reality, I do not think we will have 8, more closer to 3, though this is still being
+/// explored. This number could be larger if we consider combining different zkVMs with different guests.
+pub const MAX_EXECUTION_PROOF_SUBNETS: u64 = 8;
```

Comment:
If producers only subscribe and emit to a specific of subnets they support, and consumers only ready the set of incoming proofs they support, why do we need a max count of overall subnets?